python/ray/data/datasource/file_meta_provider.py

```diff
@@ -120,6 +120,7 @@ def expand_paths(
         raise NotImplementedError
 
 
+@DeveloperAPI
```

Comment:
Same here, this abstraction is used as an argument in the public `read*()` [APIs](https://github.com/ray-project/ray/blob/52b1e4b6722ff681f536e8df3a12754b6018e702/python/ray/data/read_api.py#L287); should this and the abstractions below be considered public APIs?